R/utility.R

```diff
@@ -62,3 +62,84 @@ get.encoded.distance <- function(points) {
 
     return(encoded.distances);
     }
+
+#' Generate a named vector of colors for every clone ID specified.
+#'
+#' Assigns colors to clones, ensuring a minimum number of colors are used.
+#' Any colors specified in `clone.colours` are maintained in the order
+#' specified and are used as the first colors for the `clone.ids`.
+#'
+#' if `clone.colours` is NULL or an empty vector, _and_
+#' `minimum.number.of.colours` is 0, `NULL` is returned
+#' so BPG's default color selection can be used without needing to
+#' check for `NULL`.
+#'
+#' @param clone.colours A vector of colors assigned to clones. If `NULL`, colors will be generated automatically.
+#' @param clone.ids A vector of clone identifiers.
+#' @param minimum.number.of.colours An integer specifying the minimum number of colors required.
+#'
+#' @return A named vector of colors assigned to each clone.
+get.clone.colours <- function(clone.colours, clone.ids, minimum.number.of.colours = 0) {
```

Comment:
Is it possible to consolidate this with `get.colours`? I think having two different colour functions are redundant and `get.colours` is already used in `create.cluster.heatmap` and `create.clone.genome.distribution.plot`

Reply:
Possibly. I didn't know there was already a `get.colours`. I will investigate and get back to you.